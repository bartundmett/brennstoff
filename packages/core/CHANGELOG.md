# @brennstoff/basetheme

## 0.1.3

### Patch Changes

- e0f2969: Fixed button props and removed dependency on basetheme

## 0.1.2

### Patch Changes

- Updated dependencies [9cd76db]
  - @brennstoff/basetheme@0.1.2

## 0.1.1

### Patch Changes

- Updated dependencies [1ee702c]
  - @brennstoff/basetheme@0.1.1

## 0.1.0

### Minor Changes

- 83b115e: Added theming of components, some helpers and a button component

### Patch Changes

- Updated dependencies [83b115e]
  - @brennstoff/basetheme@0.1.0

## 0.0.2

### Patch Changes

- c974dae: Created basic structure for package library
