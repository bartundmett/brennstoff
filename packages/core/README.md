<div align="center">
  <h1 align="center">
    Brennstoff
  </h1>
</div>

<p align="center">
<a href="https://bundlephobia.com/result?p=@brennstoff/core">
    <img alt="Bundlephobia" src="https://img.shields.io/bundlephobia/minzip/@brennstoff/core?style=for-the-badge&labelColor=24292e">
  </a>
  <a aria-label="Types" href="https://www.npmjs.com/package/@brennstoff/core">
    <img alt="Types" src="https://img.shields.io/npm/types/@brennstoff/core?style=for-the-badge&labelColor=24292e">
  </a>
<a href="https://www.npmjs.com/package/@brennstoff/core"><img src="https://img.shields.io/npm/v/@brennstoff/core?style=for-the-badge&labelColor=24292e"></a>
<a href="https://www.npmjs.com/package/@brennstoff/core"><img src="https://img.shields.io/npm/l/@brennstoff/core?style=for-the-badge&labelColor=24292e"></a>
<p>

<pre align="center">yarn add @brennstoff/core</pre>
