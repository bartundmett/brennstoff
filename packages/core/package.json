{
  "name": "@brennstoff/core",
  "version": "0.1.4",
  "description": "Core Package",
  "author": "Benedikt Grimm <benedikt.grimm@bartundmett.de>",
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/bartundmett/brennstoff.git"
  },
  "scripts": {
    "dev": "tsdx watch --tsconfig tsconfig.build.json --verbose --noClean",
    "build": "tsdx build --tsconfig tsconfig.build.json",
    "test": "tsdx test",
    "lint": "tsdx lint",
    "prepublish": "npm run build"
  },
  "main": "dist/index.js",
  "module": "dist/theme.esm.js",
  "typings": "dist/index.d.ts",
  "files": [
    "README.md",
    "dist"
  ],
  "peerDependencies": {
    "react": "^17.0.1",
    "react-dom": "^17.0.1"
  },
  "dependencies": {
    "@types/styled-components": "^5.1.7",
    "@types/styled-system": "^5.1.10",
    "@types/styled-system__theme-get": "^5.0.1",
    "@styled-system/theme-get": "^5.1.2",
    "styled-components": "^5.2.1",
    "styled-system": "^5.1.5",
    "tslib": "^2.0.0"
  },
  "devDependencies": {}
}