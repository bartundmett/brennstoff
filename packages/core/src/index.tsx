import * as React from 'react';
import {
  createGlobalStyle,
  StyledInterface,
  ThemeProvider,
} from 'styled-components';
import { variant } from 'styled-system';
import styledComponents from 'styled-components';
import * as system from 'styled-system';

export * from './forward-ref-component';
export { default as themeget } from '@styled-system/theme-get';

export const compose = system.compose;
export const space = system.space;
export const position = system.position;
export const styled: StyledInterface = styledComponents;

const GlobalStyles = createGlobalStyle`
`;

export function sharedStyles(component: string) {
  return (props: any) => props.theme.components[component].shared;
}

export function variantStyles(component: string, prop: string) {
  return variant({
    scale: `components.${component}.${prop}`,
    prop,
    variants: {},
  });
}

export function componentStyles(
  component: string,
  variants: Array<string>,
): Array<any> {
  const styles = [sharedStyles(component)];
  variants.forEach(variant => {
    styles.push(variantStyles(component, variant));
  });

  return styles;
}

export function Brennstoff({
  children,
  theme,
}: {
  children: React.ReactNode;
  theme: any;
}) {
  return (
    <ThemeProvider theme={theme}>
      <GlobalStyles />
      {children}
    </ThemeProvider>
  );
}
