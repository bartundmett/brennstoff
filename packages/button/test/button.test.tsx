import * as React from 'react';
import 'jest-axe/extend-expect';
import 'jest-styled-components';
import { axe } from 'jest-axe';
import { cleanup, fireEvent, render } from '@testing-library/react';

import Button from '../src';
import { Brennstoff } from '@brennstoff/core';
import defaultTheme from '@brennstoff/basetheme';

function Wrappers({ children }: { children: React.ReactNode }) {
  return <Brennstoff theme={defaultTheme}>{children}</Brennstoff>;
}
const customRender = (ui: JSX.Element, options: any = {}) =>
  render(ui, { wrapper: Wrappers, ...options });

describe('Button', () => {
  afterEach(cleanup);

  test('should have no accessibility violations', async () => {
    const { container, getByTestId } = customRender(
      <Button data-testid="button-test" label="" icon={<span>icon</span>} />,
    );

    fireEvent.click(getByTestId('button-test'));
    const results = await axe(container);
    expect(results).toHaveNoViolations();
    expect(container).toMatchSnapshot();
  });

  test('basic', () => {
    const { container } = customRender(<Button>Button Test</Button>);
    expect(container).toMatchSnapshot();
  });

  test('is polymorphic', () => {
    const { container } = customRender(
      <Button as="a" href="#">
        Button Test
      </Button>,
    );
    expect(container.querySelector('a')).not.toBeNull();
    expect(container).toMatchSnapshot();
  });
});
