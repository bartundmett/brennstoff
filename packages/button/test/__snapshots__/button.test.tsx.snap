// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button basic 1`] = `
.c0 {
  border: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  cursor: pointer;
  font-family: ui-sans-serif,system-ui,-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,"Helvetica Neue",Arial,"Noto Sans",sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol","Noto Color Emoji";
  font-weight: 700;
  text-transform: uppercase;
  padding-left: 1rem;
  padding-right: 1rem;
  color: #FFF;
  background-color: #FDE68A;
  font-size: 0.875rem;
  height: 48px;
}

.c0:focus {
  outline: none;
}

.c0:hover {
  background-color: #FBBF24;
}

.c0[disabled] {
  cursor: not-allowed;
  background: #D1D5DB;
  color: #6B7280;
}

<div>
  <button
    class="c0"
    type="button"
  >
    Button Test
  </button>
</div>
`;

exports[`Button is polymorphic 1`] = `
.c0 {
  border: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  cursor: pointer;
  font-family: ui-sans-serif,system-ui,-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,"Helvetica Neue",Arial,"Noto Sans",sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol","Noto Color Emoji";
  font-weight: 700;
  text-transform: uppercase;
  padding-left: 1rem;
  padding-right: 1rem;
  color: #FFF;
  background-color: #FDE68A;
  font-size: 0.875rem;
  height: 48px;
}

.c0:focus {
  outline: none;
}

.c0:hover {
  background-color: #FBBF24;
}

.c0[disabled] {
  cursor: not-allowed;
  background: #D1D5DB;
  color: #6B7280;
}

<div>
  <a
    class="c0"
    href="#"
    type="button"
  >
    Button Test
  </a>
</div>
`;

exports[`Button should have no accessibility violations 1`] = `
.c0 {
  border: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  cursor: pointer;
  font-family: ui-sans-serif,system-ui,-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,"Helvetica Neue",Arial,"Noto Sans",sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol","Noto Color Emoji";
  font-weight: 700;
  text-transform: uppercase;
  padding-left: 1rem;
  padding-right: 1rem;
  color: #FFF;
  background-color: #FDE68A;
  font-size: 0.875rem;
  height: 48px;
}

.c0:focus {
  outline: none;
}

.c0:hover {
  background-color: #FBBF24;
}

.c0[disabled] {
  cursor: not-allowed;
  background: #D1D5DB;
  color: #6B7280;
}

<div>
  <button
    aria-label=""
    class="c0"
    data-testid="button-test"
    type="button"
  >
    <span>
      icon
    </span>
  </button>
</div>
`;
